### PR TITLE
Replace xfs with ext4 for r3 SSD instances

### PR DESCRIPTION
**Pull request by rxin**
In my experience xfs had a lot of problems during shuffle heavy workloads. All the problems I observed went away when I replaced the file system with ext4. We might be able to tune xfs to be better; however, ext4 just provides better out of the box behavior here without us having to dig deeper into file system tuning.

### CODE REVIEW

File: setup-slave.sh [hunk 1]
```diff
@@ -14,27 +14,30 @@ HOSTNAME=$PRIVATE_DNS  # Fix the bash built-in hostname variable too
...
-  yum install -y xfsprogs
-
+  # Format & mount using ext4, which has the best performance among ext3, ext4, and xfs based
+  # on our shuffle heavy benchmark
+  EXT4_MOUNT_OPTS="defaults,noatime,nodiratime"
   rm -rf /mnt*
   mkdir /mnt
-  mkfs.xfs /dev/sdb
-  mount -o $XFS_MOUNT_OPTS /dev/sdb /mnt
+  #echo '/dev/sdb /mnt  ext4  defaults,noatime,nodiratime,discard 0 0' >> /etc/fstab
```
Comment (pwendell) [inline setup-slave.sh:26]: Did you mean to have this uncommented?

**Comment by pwendell** [comment]
LGTM except for one minor comment. Also, it would be good to merge the finalized version of this into v2 as well for older Spark versions

File: setup-slave.sh (hunk 1 above)
Comment (rxin) [inline setup-slave.sh:26]: Yes - it is not obvious whether we should turn TRIM support on by default. I added another line of comment to explain that.

**Comment by pwendell** [comment]
LGTM - do you mind submitting this to v2 as well?

**Comment by pwendell** [comment]
@rxin I went ahead and manually ported these changes to v2, so don't worry about it.

**Comment by rxin** [comment]
Thanks! (Sorry haven't had a chance to do it yet)

**Pull request merged**